hs-ucan/library/Web/UCAN/Validation.hs

```diff
@@ -15,164 +13,112 @@ import qualified Crypto.PubKey.RSA.PKCS15       as Crypto.RSA.PKCS
 import           RIO                            hiding (exp)
 import           RIO.Time
 
-import           Data.Aeson
-
-import           Control.Monad.Time
-
 import           Crypto.Key.Asymmetric          as Key
 import           Web.DID.Types                  as DID
-import           Web.SemVer.Types
-
-import           Web.UCAN.Resolver              as Proof
 
 import           Web.UCAN.Claims.Error
 import           Web.UCAN.Header.Error
 import           Web.UCAN.Signature.Error
 
 import           Web.UCAN                       as UCAN
 import           Web.UCAN.Error                 as UCAN
-import           Web.UCAN.Proof                 as UCAN.Proof
-
+import qualified Web.UCAN.Proof.Error           as Proof
 import qualified Web.UCAN.Signature.RS256.Types as RS256
 import           Web.UCAN.Signature.Types       as Signature
 
-check ::
```

Comment:
`check` doesn't quite work with the same interface with UCANs that can have multiple capabilities, so I didn't want to revive it.
I think we have a rough toolbox of functions that cover our needs. Today the interface isn't great, I have some ideas, but I want to start using the current tools to get a better understanding of what exactly would work well.

Reply:
Also for posterity: Philipp and I discussed a strategy for this in person, too. It's not a blocker, but could be nice to have a top-level function at some stage.